lib/m2ts/caption-stream.js

```diff
@@ -688,18 +688,32 @@ Cea708Stream.prototype.handleText = function(i, service, options) {
   var char;
   var charCodeArray;
 
+  // Converts an array of bytes to a unicode hex string.
+  function toHexString(byteArray) {
+    return byteArray.map((byte) => {
+      return ('0' + (byte & 0xFF).toString(16)).slice(-2);
+    }).join('');
+  };
+
+  if (isMultiByte) {
+    charCodeArray = [currentByte, nextByte];
+    i++;
+  } else {
+    charCodeArray = [currentByte];
+  }
+
   // Use the TextDecoder if one was created for this service
   if (service.textDecoder_ && !isExtended) {
```

Comment:
Would love feedback from the team here, as in the cases we are seeing issues, there is no `textDecoder_` set. I _think_ that this is a real use case and not an issue with the 708 data, but feel free to chime in if that is not the case.

Reply:
I agree this is a valid use case and not a 708 data issue. In order to create a textDecoder we need an `encoding` string, which it seems is missing in this case. I think it's best to try and decode the text anyway.